modules/javafx.graphics/src/main/native-decora/SSELinearConvolveShadowPeer.cc

```diff
@@ -118,6 +118,20 @@ Java_com_sun_scenario_effect_impl_sw_sse_SSELinearConvolveShadowPeer_filterHV
      jintArray srcPixels_arr, jint srccols, jint srcrows, jint scolinc, jint srowinc,
      jfloatArray kvals_arr, jfloatArray shadowColor_arr)
 {
+    if (srcPixels_arr == NULL ||
+        dstPixels_arr == NULL ||
+        srccols <= 0 ||
+        srcrows <= 0 ||
+        srccols > INT_MAX / srcrows ||
+        dstcols <= 0 ||
+        dstrows <= 0 ||
+        dstcols > INT_MAX / dstrows ||
+        (srccols * srcrows) > env->GetArrayLength(srcPixels_arr) ||
+        (dstcols * dstrows) > env->GetArrayLength(dstPixels_arr) ||
+        dstrows > srcrows) { // We should not move out of source vertical bounds
+        return;
+    }
```

Comment:
Instead of copy-pasting the same checks in all the missing places, isn't it better to create a check method (say in `SSEUtils`) and call if from all the places? Like in the style of `java.util.Objects::checkRange`.

Reply:
Thanks for the inputs.
Apart from single check all others are identical, so i have added common utility function.